include/SDL3/SDL_gpu.h

```diff
@@ -941,212 +941,373 @@ typedef enum SDL_GPUDriver
+ *
+ * \since This struct is available since SDL 3.0.0
+ *
+ * \sa SDL_DispatchGPUComputeIndirect
+ */
 typedef struct SDL_GPUIndirectDispatchCommand
 {
-    Uint32 groupcount_x;
-    Uint32 groupcount_y;
-    Uint32 groupcount_z;
+    Uint32 groupcount_x;  /**< The number of local workgroups to dispatch in the X dimension. */
+    Uint32 groupcount_y;  /**< The number of local workgroups to dispatch in the Y dimension. */
+    Uint32 groupcount_z;  /**< The number of local workgroups to dispatch in the Z dimension. */
 } SDL_GPUIndirectDispatchCommand;
 
 /* State structures */
 
+/**
+ * A structure specifying the parameters of a sampler.
+ *
+ * \since This function is available since SDL 3.0.0
+ *
+ * \sa SDL_CreateGPUSampler
+ */
 typedef struct SDL_GPUSamplerCreateInfo
 {
-    SDL_GPUFilter min_filter;
-    SDL_GPUFilter mag_filter;
-    SDL_GPUSamplerMipmapMode mipmap_mode;
-    SDL_GPUSamplerAddressMode address_mode_u;
-    SDL_GPUSamplerAddressMode address_mode_v;
-    SDL_GPUSamplerAddressMode address_mode_w;
-    float mip_lod_bias;
-    float max_anisotropy;
-    SDL_bool enable_anisotropy;
-    SDL_bool enable_compare;
+    SDL_GPUFilter min_filter;                  /**< The minification filter to apply to lookups. */
+    SDL_GPUFilter mag_filter;                  /**< The magnification filter to apply to lookups. */
+    SDL_GPUSamplerMipmapMode mipmap_mode;      /**< The mipmap filter to apply to lookups. */
+    SDL_GPUSamplerAddressMode address_mode_u;  /**< The addressing mode for U coordinates outside [0, 1). */
+    SDL_GPUSamplerAddressMode address_mode_v;  /**< The addressing mode for V coordinates outside [0, 1). */
+    SDL_GPUSamplerAddressMode address_mode_w;  /**< The addressing mode for W coordinates outside [0, 1). */
+    float mip_lod_bias;                        /**< The bias to be added to mipmap LOD calculation. */
+    float max_anisotropy;                      /**< The anisotropy value clamp used by the sampler. If enable_anisotropy is SDL_FALSE, this is ignored. */
+    SDL_bool enable_anisotropy;                /**< SDL_TRUE to enable anisotropic filtering. */
+    SDL_bool enable_compare;                   /**< SDL_TRUE to enable comparison against a reference value during lookups. */
     Uint8 padding1;
     Uint8 padding2;
-    SDL_GPUCompareOp compare_op;
-    float min_lod;
-    float max_lod;
+    SDL_GPUCompareOp compare_op;               /**< The comparison operator to apply to fetched data before filtering. */
+    float min_lod;                             /**< Clamps the minimum of the computed LOD value. */
+    float max_lod;                             /**< Clamps the maximum of the computed LOD value. */
 
-    SDL_PropertiesID props;
+    SDL_PropertiesID props;                    /**< A properties ID for extensions. Should be 0 if no extensions are needed. */
 } SDL_GPUSamplerCreateInfo;
 
+/**
+ * A structure specifying a vertex binding.
```

Comment:
Could maybe elaborate slightly on the relationship between vertex bindings and attributes?